### ISSUE TITLE
Why do I need to input class weight as class count into hinge loss?

### ISSUE THREAD

**Issue by JarvisUSTC** [richard-peng-xia/LMPT#4]
```
class_weights = sfm(torch.from_numpy(np.asarray(mmcv.load(freq_file)['neg_class_freq'])).to(torch.float32).cuda())
hinge_loss = SoftMarginHingeEmbeddingLoss(margin=0.2, class_counts=class_weights)
```

Why do I need to input class weight as class count into hinge loss? By the way, the class weights is two-dimension which is not suitable for the hinge_loss.

**Comment by richard-peng-xia** [comment]
Hi @JarvisUSTC ,

Thank you for showing interest in LMPT!

1. The reason why class weight is required in the calculation of hinge loss is because it incorporates the re-weighting strategy, which makes the model perform better when it comes to long-tailed distribution data. For details, you can refer to Eq. 5 and 6 in the [paper](https://arxiv.org/abs/2305.04536).

2. The class weight is a one-dimensional tensor. Below is an example. I hope it may help you.
```
inputs = torch.randn(3, 3)
# labels should be 1 or -1
labels = 2 * (torch.rand(3, 3) > 0.5).float() - 1 
# class_weights should be a tensor with the same shape as the labels
class_counts = torch.tensor([6, 34, 8, 14, 154, 17, 249, 29, 479, 4, 240, 48, 11, 20, 775, 100, 5, 82, 7, 64])
hinge_loss = SoftMarginHingeEmbeddingLoss(class_counts=class_counts)
loss = hinge_loss(inputs, labels)
print(loss)
```
